framework/src/main/java/org/tron/core/net/messagehandler/TransactionsMsgHandler.java

```diff
@@ -102,7 +103,7 @@ private void check(PeerConnection peer, TransactionsMessage msg) throws P2pExcep
   private void handleSmartContract() {
     smartContractExecutor.scheduleWithFixedDelay(() -> {
       try {
-        while (queue.size() < MAX_SMART_CONTRACT_SUBMIT_SIZE) {
+        while (queue.size() < MAX_SMART_CONTRACT_SUBMIT_SIZE && smartContractQueue.size() > 0) {
```

Comment:
Try not to change the original logic, if it is a bug, it is recommended to submit a pr separately.

Reply:
@wubin01 , As explained above,  `smartContractExecutor.shutdown()` -> `ExecutorServiceManager.shutdownAndAwaitTermination(smartContractExecutor, smartEsName)`
if queue is empty, will block 60s